modules/map/tools/map_datachecker/client/client.cc

```diff
@@ -0,0 +1,250 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#include "modules/map/tools/map_datachecker/client/client.h"
+
+#include <yaml-cpp/yaml.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <string>
+#include <vector>
+
+#include "cyber/cyber.h"
+#include "modules/map/tools/map_datachecker/client/client_alignment.h"
+#include "modules/map/tools/map_datachecker/client/client_channel_checker.h"
+#include "modules/map/tools/map_datachecker/client/client_common.h"
+#include "modules/map/tools/map_datachecker/client/client_loops_check.h"
+
+namespace apollo {
+namespace hdmap {
+
+Client::Client() {
+  YAML::Node node = YAML::LoadFile(FLAGS_client_conf_yaml);
+  std::string bin_path = boost::filesystem::current_path().string();
+  data_collect_time_flag_file_ =
+      bin_path + "/" + node["time_flag_file"].as<std::string>();
+  channel_checker_stop_flag_file_ =
+      bin_path + "/" +
+      node["channel_check"]["stop_flag_file"].as<std::string>();
+  AINFO << "bin_path: " << bin_path
+        << ", data_collect_time_flag_file_: " << data_collect_time_flag_file_
+        << ", channel_checker_stop_flag_file_: "
+        << channel_checker_stop_flag_file_;
+}
+
+int Client::Run() {
+  std::string stage = FLAGS_stage;
+  AINFO << "stage [" << stage << "]";
+  if ("record_check" == stage) {
+    RecordCheckStage();
+  } else if ("static_align" == stage) {
+    StaticAlignStage();
+  } else if ("eight_route" == stage) {
+    EightRouteStage();
+  } else if ("data_collect" == stage) {
+    DataCollectStage();
+  } else if ("loops_check" == stage) {
+    LoopsCheckStage();
+  } else if ("clean" == stage) {
+    CleanStage();
+  }
+  return 0;
```

Comment:
Each function in the if has its own return value, instead of return 0 for all cases. Don't you have to return the function return values for each case?

Reply:
Ok